extensions/objects/models/objects.py

```diff
@@ -1633,208 +1552,265 @@ def get_schema(cls):
 
-        return raw
+        return value
 
 
 class TranslatableSetOfUnicodeString(BaseTranslatableObject):
     """Class for translatable sets of UnicodeStrings."""
 
     _value_key_name = 'unicodeStrSet'
     _value_schema = SetOfUnicodeString.get_schema()
-
     default_value = {
         'contentId': None,
-        'unicodeStrSet': []
+        'unicodeStrSet': [],
     }
 
     @staticmethod
-    def _normalize_value(raw):
-        """Validates and normalizes the value fields of the translatable object.
+    def normalize_value(value):
+        """Normalizes the translatable value of the object.
 
         Args:
-            raw: *. A translatable Python object whose values are to be
-                normalized.
+            value: *. The translatable part of the Python object (corresponding
+                to the non-content-id field) which is to be normalized.
 
         Returns:
-            dict. A normalized translatable Python object with its values
-            normalized.
-
-        Raises:
-            TypeError. The Python object cannot be normalized.
+            list(str). The normalized value.
         """
-        if not isinstance(raw['unicodeStrSet'], list):
-            raise TypeError(
-                'Invalid unicode string set: %s' % raw['unicodeStrSet'])
+        if not isinstance(value, list):
+            raise TypeError('Invalid unicode string set: %s' % value)
 
-        for unicode_str in raw['unicodeStrSet']:
-            if not isinstance(unicode_str, python_utils.BASESTRING):
+        for normalized_str in value:
+            if not isinstance(normalized_str, python_utils.BASESTRING):
                 raise TypeError(
-                    'Invalid content unicode: %s' % unicode_str)
-
-        if len(set(raw['unicodeStrSet'])) != len(raw['unicodeStrSet']):
-            raise TypeError(
-                'Duplicate unicode found in set: %s' % raw['unicodeStrSet'])
-
-        return raw
-
-
-class TranslatableHtmlContentId(BaseObject):
-    """A TranslatableHtml content id."""
-
-    default_value = ''
-
-    @classmethod
-    def get_schema(cls):
-        """Returns the object schema.
-
-        Returns:
-            dict. The object schema.
-        """
-        return UnicodeString.get_schema()
-
-
-class SetOfTranslatableHtmlContentIds(BaseObject):
-    """A Set of TranslatableHtml content ids."""
-
-    default_value = []
-
-    @classmethod
-    def get_schema(cls):
-        """Returns the object schema.
-
-        Returns:
-            dict. The object schema.
-        """
-        return {
-            'type': 'list',
-            'items': TranslatableHtmlContentId.get_schema(),
-            'validators': [{
-                'id': 'is_uniquified'
-            }]
-        }
+                    'Invalid content unicode: %s' % normalized_str)
 
+        if len(set(value)) != len(value):
+            raise TypeError('Duplicate unicode found in set: %s' % value)
```

Comment:
should we add `is_unique` validator in the schema and the schema_utils will handle it?

Reply:
Also simplified as above -- thanks!